src/App.tsx

```diff
@@ -1,204 +1,101 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import { Errors } from './types/Errors';
+import { Status } from './types/Status';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+const USER_ID = 96;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(Status.all);
+  //  const [visibleTodos, setVisibleTodos] = useState(todos);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+
+  const onAllClick = useCallback(() => {
+    if (filterBy !== Status.all) {
+      setFilterBy(Status.all);
+    }
+  }, [filterBy]);
+
+  const onActiveClick = useCallback(() => {
+    if (filterBy !== Status.active) {
+      setFilterBy(Status.active);
+    }
+  }, [filterBy]);
+
+  const onCompletedClick = useCallback(() => {
+    if (filterBy !== Status.completed) {
+      setFilterBy(Status.completed);
+    }
+  }, [filterBy]);
+
+  const visibleTodos = useMemo(() => {
+    switch (filterBy) {
+      case Status.all:
+        return todos;
+
+      case Status.active:
+        return todos.filter(todo => !todo.completed);
+
+      case Status.completed:
+        return todos;
+
+      default:
+        return todos;
+    }
+  }, [filterBy, todos]);
+  const hideError = useCallback(() => {
+    setErrorMessage(null);
+  }, []);
+
+  useEffect(() => {
+    hideError();
+
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(Errors.loadError);
+      });
+  }, [hideError]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <div className="todoapp">
-      <h1 className="todoapp__title">todos</h1>
-
-      <div className="todoapp__content">
-        <header className="todoapp__header">
-          {/* this buttons is active only if there are some active todos */}
-          <button
-            type="button"
-            className="todoapp__toggle-all active"
-            data-cy="ToggleAllButton"
-          />
-
-          {/* Add a todo on form submit */}
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
+    <>
+      <div className="todoapp">
+        <h1 className="todoapp__title">todos</h1>
+
+        <div className="todoapp__content" />
```

Comment:
why is this `div` closed?

I guess it should be like a container for `Header` and other components